lib/model/forum/ForumDradenMeldingModel.class.php

```diff
@@ -0,0 +1,192 @@
+		$melding->draad_id = $draad_id;
+		$melding->uid = $uid;
+		$melding->niveau = $niveau;
+		$this->create($melding);
+		return $melding;
+	}
+
+	public function getAltijdMeldingVoorDraad(ForumDraad $draad) {
+		return $this->prefetch('draad_id = ? AND niveau = ?', array($draad->draad_id, ForumDraadMeldingNiveau::ALTIJD));
+	}
+
+	public function getVoorkeursNiveauVoorLid(ForumDraad $draad, $uid = null) {
+		if ($uid === null) $uid = LoginModel::getUid();
+
+		/** @var ForumDraadMelding $voorkeur */
+		$voorkeur = $this->retrieveByPrimaryKey(array($draad->draad_id, $uid));
+		if ($voorkeur) {
+			return $voorkeur->niveau;
+		} else {
+			return false;
+		}
+	}
+
+	public function getNiveauVoorLid(ForumDraad $draad, $uid = null) {
+		if ($uid === null) $uid = LoginModel::getUid();
+
+		/** @var ForumDraadMelding $voorkeur */
+		$voorkeur = $this->retrieveByPrimaryKey(array($draad->draad_id, $uid));
+		if ($voorkeur) {
+			return $voorkeur->niveau;
+		} else {
+			$wilMeldingBijVermelding = LidInstellingenModel::getInstellingVoorLid('forum', 'meldingStandaard', $uid);
+			return $wilMeldingBijVermelding === 'ja' ? ForumDraadMeldingNiveau::VERMELDING : ForumDraadMeldingNiveau::NOOIT;
+		}
+	}
+
+	public function setNiveauVoorLid(ForumDraad $draad, $niveau) {
+		$uid = LoginModel::getUid();
+		/** @var ForumDraadMelding $voorkeur */
+		$voorkeur = $this->retrieveByPrimaryKey(array($draad->draad_id, $uid));
+		if ($voorkeur) {
+			$voorkeur->niveau = $niveau;
+			$this->update($voorkeur);
+		} else {
+			$this->maakForumDraadMelding($draad->draad_id, $uid, $niveau);
+		}
+	}
+
+	public function stopAlleMeldingenVoorLid($uid) {
+		foreach ($this->find('uid = ?', array($uid)) as $volgen) {
+			$this->delete($volgen);
+		}
+	}
+
+	public function stopMeldingenVoorIedereen($draad) {
+		foreach ($this->find('draad_id = ?', array($draad->draad_id)) as $volgen) {
+			$this->delete($volgen);
+		}
+	}
+
+	public function stuurMeldingen(ForumPost $post) {
+		$this->stuurMeldingenNaarVolgers($post);
+		$this->stuurMeldingenNaarGenoemden($post);
+	}
+
+	/**
+	 * Verzendt mail
+	 *
+	 * @param Profiel $ontvanger
+	 * @param Profiel $auteur
+	 * @param ForumPost $post
+	 * @param ForumDraad $draad
+	 * @param string $bericht
+	 */
+	private function stuurMelding($ontvanger, $auteur, $post, $draad, $bericht) {
+		$ontvangerNaam = $ontvanger->getNaam('civitas');
+		$values = array(
+			'NAAM' => $ontvangerNaam,
+			'AUTEUR' => $auteur->getNaam('civitas'),
+			'POSTLINK' => $post->getLink(true),
+			'TITEL' => $draad->titel,
+			'TEKST' => str_replace('\r\n', "\n", $post->tekst),
+		);
+
+		// Stel huidig UID in op ontvanger om te voorkomen dat ontvanger privé of andere persoonlijke info te zien krijgt
+		$curUID = $_SESSION['_uid'];
+		$_SESSION['_uid'] = $ontvanger->uid;
+
+		// Verzend mail
+		$mail = new Mail(array($ontvanger->getPrimaryEmail() => $ontvangerNaam), 'C.S.R. Forum: nieuwe reactie op ' . $draad->titel, $bericht);
+		$mail->setPlaceholders($values);
```

Comment:
Het is beter om hier `getNaam('volledig')` te gebruiken, want er zijn een paar gare spam filters die stuk gaan van • in een naam.